parastell/utils.py

```diff
@@ -1,70 +1,92 @@
 
 
-def expand_ang_list(ang_list, num_ang):
-    """Expands list of angles by linearly interpolating according to specified
-    number to include in stellarator build.
+def enforce_helical_symmetry(matrix):
+    """Ensures that a matrix is helically symmetric according to stellarator
+    geometry by overwriting certain matrix elements. Assumes regular spacing
+    between angles defining matrix.
 
     Arguments:
-        ang_list (list of double): user-supplied list of toroidal or poloidal
-            angles (rad).
-        num_ang (int): number of angles to include in stellarator build.
+        matrix (2-D iterable of float): matrix to be made helically symmetric.
 
     Returns:
-        ang_list_exp (list of double): interpolated list of angles (rad).
+        matrix (2-D iterable of float): helically symmetric matrix.
     """
-    ang_list = np.deg2rad(ang_list)
+    num_rows = matrix.shape[0]
+    num_columns = matrix.shape[1]
+
+    # Ensure poloidal symmetry at beginning and middle of period
+    matrix[0] = np.concatenate(
+        [
+            matrix[0, : int((num_columns - 1) / 2) + 1],
+            np.flip(matrix[0, : int(num_columns / 2)]),
+        ]
+    )
+    matrix[int((num_rows - 1) / 2)] = np.concatenate(
+        [
+            matrix[int((num_rows - 1) / 2), : int((num_columns - 1) / 2) + 1],
+            np.flip(matrix[int((num_rows - 1) / 2), : int(num_columns / 2)]),
+        ]
+    )
+
+    # Ensure helical symmetry toroidally and poloidally
+    for index in range(num_rows - 1, int((num_rows - 1) / 2), -1):
+        matrix[num_rows - 1 - index, -1] = matrix[num_rows - 1 - index, 0]
+        matrix[index] = np.flip(matrix[num_rows - 1 - index])
+
+    return matrix
+
+
+def expand_list(list, num):
+    """Expands a list of ordered floats to a total number of entries by
+    linearly interpolating between entries, inserting a proportional number of
+    new entries between original entries.
 
-    ang_list_exp = []
-
-    init_ang = ang_list[0]
-    final_ang = ang_list[-1]
-    ang_extent = final_ang - init_ang
-
-    ang_diff_avg = ang_extent / (num_ang - 1)
+    Arguments:
+        list (iterable of float): list to be expanded.
+        num (int): desired number of entries in expanded list.
 
-    for ang, next_ang in zip(ang_list[:-1], ang_list[1:]):
-        n_ang = math.ceil((next_ang - ang) / ang_diff_avg)
+    Returns:
+        list_exp (iterable of float): expanded list.
+    """
+    list_exp = []
 
-        ang_list_exp = np.append(
-            ang_list_exp, np.linspace(ang, next_ang, num=n_ang + 1)[:-1]
-        )
+    init_entry = list[0]
+    final_entry = list[-1]
+    extent = final_entry - init_entry
 
-    ang_list_exp = np.append(ang_list_exp, ang_list[-1])
+    avg_diff = extent / (num - 1)
 
-    return ang_list_exp
+    for entry, next_entry in zip(list[:-1], list[1:]):
+        num_new_entries = int(round((next_entry - entry) / avg_diff))
 
+        list_exp = np.append(
```

Comment:
Subtle feature may warrant a comment
```suggestion
        # don't append the last entry in the created linspace to avoid adding it twice when
        # the next created linspace is appended
        list_exp = np.append(
```